### PR TITLE
Add: New File, Utility Count, Balancing Authority Count

### LINKED ISSUE #20: Add count: # electric utilities operating with watersheds

### LINKED ISSUE #30: Update data for power plants
UCS data to be supplanted by more up to date power plant data that includes balancing authorities for each plant.

EIA - power plants and balancing authorities (@swd-turner to locate this).
Ryan's power plant data [here](https://www.sciencebase.gov/catalog/item/58209752e4b080404e6fa8f0).

### PR DESCRIPTION
**Pull request by KristianNelson**
This change resolves #30 and resolves #20 . Adds a new file for the utilities and BAs, merges that to UCS, and the counts the utilities and BAs.

### CODE REVIEW

File: R/io.R [hunk 1]
```diff
@@ -139,18 +139,30 @@ get_crop_mapping <- function(){
...
 get_ucs_power_plants <- function(ucs_file_path,
                                  method = "sp"){
+
+  vroom(paste0(system.file("extdata", package = "teleconnect"),
+               "/utility_ba_data.csv"),
+        col_types = cols(Utility_ID = col_double(),
+                         Utility_Name = col_character(),
+                         Plant_Code = col_double(),
+                         Balancing_Authority = col_character()),
+        skip = 1) -> utility_ba_data
```
Comment (swd-turner) [inline R/io.R]: Please provide a `delim = ","` argument to vroom, as this fails on my machine.

File: R/io.R [hunk 2]
```diff
@@ -139,18 +139,30 @@ get_crop_mapping <- function(){
...
     select(cooling = `Requires cooling?`,
            cooling_tech = `Cooling Technology`,
+           Plant_Code = `Plant Code`,
            `Power Plant Type` = Fuel,
            `Nameplate Capacity (MW)`,
-           lat = Latitude, lon = Longitude) ->
-    ucs_plants
+           lat = Latitude, lon = Longitude) -> ucs
+
+  merge(ucs, utility_ba_data, by = "Plant_Code") -> ucs_plants
```
Comment (swd-turner) [inline R/io.R]: Use `left_join`, not `merge`

**Comment by KristianNelson** [comment]
I do want to mention that if you're seeing a lot of things like `dplyr::` or `stringr::` behind arguments, that's because I cannot get it to run without having those there even when they're defined at the top. If there is something that I am not doing correctly let me know.

**Comment by swd-turner** [comment]
Hey @KristianNelson the reason these functions don't work is that you've not updated the documentation to bring in those new imports. In RStudio go to the `Build` tab, then `More` and `Document`. The way you've done it is fine for now... we can clean up later.

**Pull request merged**